src/anemoi/inference/post_processors/extract.py

```diff
@@ -0,0 +1,122 @@
+# (C) Copyright 2024-2025 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+import logging
+
+import numpy as np
+
+from anemoi.inference.context import Context
+from anemoi.inference.decorators import main_argument
+from anemoi.inference.types import BoolArray
+from anemoi.inference.types import State
+
+from ..processor import Processor
+from . import post_processor_registry
+
+LOG = logging.getLogger(__name__)
+
+
+class ExtractBase(Processor):
+    """Base class for processors that extract data from the state."""
+
+    indexer: BoolArray | slice
+
+    def process(self, state: State) -> State:
+        """Process the state to extract a subset of points based on the indexer.
+
+        Parameters
+        ----------
+        state : State
+            The state containing fields to be extracted.
+
+        Returns
+        -------
+        State
+            The updated state with extracted fields.
+        """
+        state = state.copy()
+
+        state["latitudes"] = state["latitudes"][self.indexer]
+        state["longitudes"] = state["longitudes"][self.indexer]
+        for field in state["fields"]:
+            state["fields"][field] = state["fields"][field][self.indexer]
+
+        return state
+
+
+@post_processor_registry.register("extract_mask")
+@main_argument("mask")
+class ExtractMask(ExtractBase):
+    """Extract a subset of points from the state based on a boolean mask.
+
+    Parameters
+    ----------
+    context : Any
+        The context in which the processor is running.
+    mask : str
+        Either a path to a `.npy` file containing the boolean mask or
+        the name of a supporting array found in the checkpoint.
+    """
+
+    def __init__(self, context: Context, mask: str) -> None:
+        super().__init__(context)
+
+        self._maskname = mask
+
+        if mask.endswith(".npy"):
```

Comment:
Instead of checking the extension, maybe something more general like `Path(mask).exists()` ?